Fix depreciations in node-keytar v4.3.0

When building node-keytar, I have this two depreciations multiple time.

```
CXX(target) Release/obj.target/keytar/src/async.o
In file included from ../src/async.cc:4:
../../nan/nan.h:839:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/test/.electron-gyp/.node-gyp/iojs-3.0.2/src/node.h:93:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/test/.electron-gyp/.node-gyp/iojs-3.0.2/src/core.h:35:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
```

```
../src/main.cc:8:6: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
    *v8::String::Utf8Value(info[0]),
     ^
/Users/test/.electron-gyp/.node-gyp/iojs-3.0.2/deps/v8/include/v8.h:2819:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/test/.electron-gyp/.node-gyp/iojs-3.0.2/deps/v8/include/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
``

--- a/package.json
+++ b/package.json
@@ -51,7 +51,7 @@
     "prebuild": "^8.1.2"
   },
   "dependencies": {
-    "nan": "2.8.0",
+    "nan": "2.12.1",
     "@shiftkey/prebuild-install": "5.2.4"
   }
 }

--- a/src/async.cc
+++ b/src/async.cc
@@ -69,7 +69,7 @@ void GetPasswordWorker::HandleOKCallback() {
     val
   };
 
-  callback->Call(2, argv);
+  callback->Call(2, argv, async_resource);
 }
 
 
@@ -105,7 +105,7 @@ void DeletePasswordWorker::HandleOKCallback() {
     val
   };
 
-  callback->Call(2, argv);
+  callback->Call(2, argv, async_resource);
 }
 
 
@@ -144,7 +144,7 @@ void FindPasswordWorker::HandleOKCallback() {
     val
   };
 
-  callback->Call(2, argv);
+  callback->Call(2, argv, async_resource);
 }
 
 
@@ -201,12 +201,12 @@ void FindCredentialsWorker::HandleOKCallback() {
       Nan::Null(),
       val
     };
-    callback->Call(2, argv);
+    callback->Call(2, argv, async_resource);
   } else {
     v8::Local<v8::Value> argv[] = {
       Nan::Null(),
       Nan::New<v8::Array>(0)
     };
-    callback->Call(2, argv);
+    callback->Call(2, argv, async_resource);
   }
 }

--- a/src/main.cc
+++ b/src/main.cc
@@ -4,40 +4,63 @@
 namespace {
 
 NAN_METHOD(SetPassword) {
+  Nan::Utf8String serviceNan(info[0]);
+  std::string service(*serviceNan, serviceNan.length());
+  Nan::Utf8String usernameNan(info[1]);
+  std::string username(*usernameNan, usernameNan.length());
+  Nan::Utf8String passwordNan(info[2]);
+  std::string password(*passwordNan, passwordNan.length());
+
   SetPasswordWorker* worker = new SetPasswordWorker(
-    *v8::String::Utf8Value(info[0]),
-    *v8::String::Utf8Value(info[1]),
-    *v8::String::Utf8Value(info[2]),
+    service,
+    username,
+    password,
     new Nan::Callback(info[3].As<v8::Function>()));
   Nan::AsyncQueueWorker(worker);
 }
 
 NAN_METHOD(GetPassword) {
+  Nan::Utf8String serviceNan(info[0]);
+  std::string service(*serviceNan, serviceNan.length());
+  Nan::Utf8String usernameNan(info[1]);
+  std::string username(*usernameNan, usernameNan.length());
+
   GetPasswordWorker* worker = new GetPasswordWorker(
-    *v8::String::Utf8Value(info[0]),
-    *v8::String::Utf8Value(info[1]),
+    service,
+    username,
     new Nan::Callback(info[2].As<v8::Function>()));
   Nan::AsyncQueueWorker(worker);
 }
 
 NAN_METHOD(DeletePassword) {
+  Nan::Utf8String serviceNan(info[0]);
+  std::string service(*serviceNan, serviceNan.length());
+  Nan::Utf8String usernameNan(info[1]);
+  std::string username(*usernameNan, usernameNan.length());
+
   DeletePasswordWorker* worker = new DeletePasswordWorker(
-    *v8::String::Utf8Value(info[0]),
-    *v8::String::Utf8Value(info[1]),
+    service,
+    username,
     new Nan::Callback(info[2].As<v8::Function>()));
   Nan::AsyncQueueWorker(worker);
 }
 
 NAN_METHOD(FindPassword) {
+  Nan::Utf8String serviceNan(info[0]);
+  std::string service(*serviceNan, serviceNan.length());
+
   FindPasswordWorker* worker = new FindPasswordWorker(
-    *v8::String::Utf8Value(info[0]),
+    service,
     new Nan::Callback(info[1].As<v8::Function>()));
   Nan::AsyncQueueWorker(worker);
 }
 
 NAN_METHOD(FindCredentials) {
+  Nan::Utf8String serviceNan(info[0]);
+  std::string service(*serviceNan, serviceNan.length());
+
   FindCredentialsWorker* worker = new FindCredentialsWorker(
-    *v8::String::Utf8Value(info[0]),
+    service,
     new Nan::Callback(info[1].As<v8::Function>()));
   Nan::AsyncQueueWorker(worker);
 }

